framework/aster-frame/src/task/processor.rs

```diff
@@ -117,9 +119,39 @@ fn switch_to_task(next_task: Arc<Task>) {
 
     let next_task_cx_ptr = &next_task.inner_ctx() as *const TaskContext;
 
+    // If the current and next tasks meet the following conditions,
+    // then we can safely skip switching page table for improved performance.
+    //
+    // Current task         | Next task                 | Should switch page table?
+    // ----------------------------------------------------------------------------
+    // any                  | w/o a user space          | No
+    // w/ a user space.     | w/ the same user space    | No
+    //
+    // In the table above, w/ is short for with and w/o is for without.
+    let should_switch_mm = match next_task.user_space() {
+        None => false,
+        Some(next_user_space) => {
+            let mut switch_mm = true;
+            if let Some(ref cur_task) = current_task {
+                if cur_task
+                    .user_space()
+                    .is_some_and(|cur_userspace| Arc::ptr_eq(cur_userspace, next_user_space))
+                {
+                    switch_mm = false;
+                }
+            }
+            switch_mm
+        }
+    };
```

Comment:
I was just wondering if the code would be clearer if we changed the type of `should_switch_mm` to an `Option<&VmSpace>`. For example, we can write:
```rust
    let vm_to_switch = next_task
        .user_space()
        .filter(|next_user_space| {
            !current_task
                .and_then(|task| task.user_space())
                .is_some_and(|cur_user_space| Arc::ptr_eq(cur_userspace, next_user_space))
        })
        .map(|next_user_space| next_user_space.vm_space());

    // ...

    if let Some(vm_space) = vm_to_switch {
        vm_space.activate();
    }
```

But it seems that the current code in this PR is clear enough, so the clarity is not really improved too much. So feel free to ignore this comment if you don't think doing so is better.

Reply:
I prefer your coding style and update accordingly.